test/selenium/src/tests/test_object_mapping.py

```
@@ -0,0 +1,79 @@
+# Copyright (C) 2015 Google Inc., authors, and contributors <see AUTHORS file>
+# Licensed under http://www.apache.org/licenses/LICENSE-2.0 <see LICENSE file>
+# Created By: jernej@reciprocitylabs.com
+# Maintained By: jernej@reciprocitylabs.com
+
+"""Tests relevant to object mapping"""
+# pylint: disable=no-self-use
+# pylint: disable=invalid-name
+# pylint: disable=too-few-public-methods
+# pylint: disable=unused-argument
+
+import pytest    # pylint: disable=import-error
+
+from lib import base
+from lib import factory
+from lib.utils import conftest_utils
+from lib.utils import selenium_utils
+from lib.page import dashboard
+
+
+class TestObjectMapping(base.Test):
+  """A part of smoke tests, section 4."""
+
+  @pytest.mark.smoke_tests
+  def test_mapping_via_lhn(self, selenium, new_product,
+                           new_project, new_system, new_data_asset,
+                           new_process, new_issue):
+    """LHN mapping tests from smoke tests, section 6"""
+    objects = [new_product, new_project, new_system, new_data_asset,
+               new_process, new_issue]
+
+    for object_ in objects:
+      selenium.driver.get(object_.url)
+      header = dashboard.Header(selenium.driver)
+
+      # map objects
+      for mapped_object in objects:
+        # don't map an object to itself
+        if mapped_object != object_:
```

If the app allows mapping an object to itself your test should probably throw an error.